packages/ra-ui-materialui/src/auth/Login.tsx

```diff
@@ -74,86 +62,72 @@ const styles = (theme: Theme) =>
  *        </Admin>
  *     );
  */
-class Login extends Component<
-    Props & WithStyles<typeof styles> & HtmlHTMLAttributes<HTMLDivElement>
-> {
-    theme = createMuiTheme(this.props.theme);
-    containerRef = React.createRef<HTMLDivElement>();
-    backgroundImageLoaded = false;
+const Login: React.SFC<Props & HtmlHTMLAttributes<HTMLDivElement>> = ({
```

Comment:
Import `FunctionComponent` from `react` and use it instead of `React.SFC` which is deprecated. They are both alias to the same type but `FunctionComponent` is now the official one (see https://github.com/DefinitelyTyped/DefinitelyTyped/pull/30364)